Move Theme Loading into Inversify Lifecycle

colin-grant-work opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md

Note: Security vulnerabilities should not be disclosed on GitHub, through a PR or any
other means. See SECURITY.md at the root of this repository, to learn how to report
vulnerabilities.
-->

#### What it does
<!-- Include relevant issues and describe how they are addressed. -->

At the moment, the initialization of theming happens outside of the application's `inversify` lifecycle. That makes it difficult to override and difficult to integrate into the management of `monaco`'s initialization lifecycle. Part of the reason for attempting to complete theme initialization outside of the `inversify` lifecycle is to ensure that themes applied accurately even to the loading screen. That seems unnecessary to me: the point of a loading screen is to give the application time to do its work. This PR sets the background of the loading screen to a constant.

#### How to test
<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->

1. Changing themes should work as before.
2. On restart / reload, theme restoration should work as before (with the caveat that the loading screen will be dark regardless of restored theme).

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

msujew reviewed (approved):
@colin-grant-work I'm wondering whether we can reintroduce storing the theme color for the loading screen later. I guess it is out of scope for this PR. As far I can tell you also are interested in it for https://github.com/eclipse-theia/theia/issues/10345, right?

Anyway, regarding the changes: I'm quite happy with them. Simplify my current work on https://github.com/eclipse-theia/theia/issues/11063 a bit 👍 As there's no noticeable regression, I'll approve this.

colin-grant-work commented on packages/core/src/browser/shell/shell-layout-restorer.ts:
```diff
@@ -145,7 +143,7 @@ export class ShellLayoutRestorer implements CommandContribution {
             this.shouldStoreLayout = false;
             this.storageService.setData(this.storageKey, undefined);
-            ThemeService.get().reset(); // Theme service cannot use DI, so the current theme ID is stored elsewhere. Hence the explicit reset.
+            this.themeService.reset(); // Theme service cannot use DI, so the current theme ID is stored elsewhere. Hence the explicit reset.
```
Good point. It is not. I'll remove it. 👍

msujew commented:
> Would you rather keep the dynamic background, then?

Yeah, I think that would be preferable, if it's not too much work. Maybe something that could be solved with the `preloader.ts`? ;)

colin-grant-work commented:
@msujew, I have brought back the code to save and load the theme color between sessions, and modified some of the startup code surrounding it to try to avoid theme flashes.

msujew reviewed (changes requested):
When loading Theia for the first time with the newest changes (can be triggered manually by removing all `idb` and `localStorage` data), theme data isn't applied for some reason:

![image](https://user-images.githubusercontent.com/4377073/173088013-651efc95-e1c3-4f9b-b752-dfa8cf7191c4.png)

With the previous commit everything works correctly

colin-grant-work commented:
> When loading Theia for the first time with the newest changes (can be triggered manually by removing all `idb` and `localStorage` data), theme data isn't applied for some reason:

Thanks for catching this. I have tried to fix the startup routine here for the different cases.
1. If the `ThemeServiceWithDB` loads the theme from its database rather than its `theme` object, `setCurrentTheme` was previously failing, because it expected the theme to be in the `theme` object unconditionally. I have added code to add the theme from the database to the `theme` object.
2. If no theme was set in local storage, then the entire `loadUserTheme` routine ran before the `ColorApplicationContribution` attached its listener for theme changes, so its `update` method was never called. I have added code that calls `update` when the `ThemeService` declares that it has completed the `loadUserTheme` routine.

colin-grant-work commented:
@msujew, sounds good. I'll do it now.
